src/core/qgsunittypes.h

```diff
@@ -95,6 +95,7 @@ class CORE_EXPORT QgsUnitTypes
     enum RenderUnit
     {
       RenderMillimeters = 0, //!< Millimeters
+      RenderMetersInMapUnits, //!< Meters value as Map units
```

Comment:
Can you move this new enum value to the end? There's a chance that code somewhere is just converting a unit to an int and saving it in the project or settings somewhere. Better to be safe and avoid changing the existing values.